pkg/network/tracer/tracer_test.go

```diff
@@ -1058,6 +1059,15 @@ func TestDNSStatsForTimeout(t *testing.T) {
 	testDNSStats(t, "golang.org", 0, 0, 1, "1.2.3.4")
 }
 
+func TestDNSStatsWithNAT(t *testing.T) {
+	// Setup a NAT rule to translate 2.2.2.2 to 8.8.8.8 and issue a DNS request to 2.2.2.2
+	cmds := []string{"iptables -t nat -A OUTPUT -d 2.2.2.2 -j DNAT --to-destination 8.8.8.8"}
+	nettestutil.RunCommands(t, cmds, true)
+	testDNSStats(t, "golang.org", 1, 0, 0, "2.2.2.2")
+	cmds = []string{"iptables -t nat -D OUTPUT -d 2.2.2.2 -j DNAT --to-destination 8.8.8.8"}
+	nettestutil.RunCommands(t, cmds, true)
```

Comment:
You probably want this in a `defer`, so it gets torn down regardless of test result.

Reply:
Done